add support for ovs offload

Mmduh-483 opened the pull request.

srikartati committed:
Fix network policy overlapping issue

Behavior with network policies with overlapping From/To blocks
is not as expected. Behavior changes with order change in addition of
policies, which is not expected. This is due to priority clash between
conjunctive match flows with overlapping match fields. Fixed this by
adding match flows for only diff addresses between From/To and Except
blocks, and not adding drop flows at all.

Added new package iputils to compute diff CIDRs.

Added a test in netpol covering policies with overlapping From/To CIDRs and
overlapping except CIDRs

Fixes: #367

antoninbas committed:
Update CHANGELOG for v0.6.0 release

Mmduh-483 committed:
add support for ovs offload

antoninbas committed:
Set VERSION to v0.7.0-dev

Mmduh-483 commented:
@moshe010

Dyanngg committed:
Add k8s version support for eks script and fix Jenkins script issues

poozza committed:
Update README.md (#623)

Changed the Roadmap section

ksamoray committed:
Register antrea_agent_local_pod_count metric (#672)

antrea_agent_local_pod_count has not been registered and therefore
missing from Promehteus client output.

jianjuns committed:
Add comments about using "tp_dst" in flow matching string (#673)

moshe010 commented on go.mod:
```diff
@@ -3,6 +3,7 @@ module github.com/vmware-tanzu/antrea
 go 1.13
 
 require (
+	github.com/Mellanox/sriovnet v0.0.0-20191103093832-a16b5a5bd280
```
please use tag v1.0.1

moshe010 commented on go.mod:
```diff
@@ -26,7 +27,7 @@ require (
 	github.com/kevinburke/ssh_config v0.0.0-20190725054713-01f96b0aa0cd
 	github.com/prometheus/client_golang v0.9.3-0.20190127221311-3c4408c8b829
-	github.com/satori/go.uuid v1.2.0
+	github.com/satori/go.uuid v1.2.1-0.20181028125025-b2ce2384e17b // indirect
```
run go mod tidy will remove this

moshe010 reviewed (changes requested):
how the validate interface works for you?
https://github.com/Mmduh-483/antrea/blob/master/pkg/agent/cniserver/pod_configuration.go#L191

moshe010 commented:
also see if we can add unit-test to https://github.com/moshe010/antrea/blob/master/pkg/agent/cniserver/server_test.go

suwang48404 committed:
Added design doc for policy-only mode. (#540)

* Added design doc for policy-only mode.

* Update docs/policy-only.md

Co-authored-by: Jianjun Shen <shenj@vmware.com>

antoninbas committed:
Update troubleshooting instructions for antrea-agent API

A bearer token is required for local access now that the API is
externally accessible (in order to expose Prometheus metrics). The
instructions to connect directly to the antrea-agent API therefore
needed to be updated.

Thanks to @srikartati for pointing out this issue to me.

weiqiangt committed:
Add namespace parameter to `podWaitForIP` (#682)

This CL makes `podWaitForIP` more generic, e.g. get IP
address of an antrea component.

Signed-off-by: Weiqiang TANG <weiqiangt@vmware.com>

jianjuns committed:
Do not add tailing spaces to rows in antctl single column output table (#676)

This is for the case rows of a single column table can be too long to
fit into one line. One example is the ovsflows outputs (every row is
an OVS flow string).

jianjuns committed:
Add Hybrid, NoEncap, NetworkPolicyOnly modes to architecture.md (#679)

* Add Hybrid, NoEncap, NetworkPolicyOnly modes to architecture.md

* Update docs/architecture.md

Co-authored-by: Abhishek Raut <rauta@vmware.com>

Co-authored-by: Abhishek Raut <rauta@vmware.com>

jianjuns committed:
Document antctl troubleshooting commands (#675)

tnqn committed:
Fix incorrect antctl example and output (#687)

* Dump OVS flows of a flow Table should be "-T <TABLE>"
* NodeSubnet should be in agentinfo

tnqn committed:
Fix flaky unit test TestAddNetworkPolicyWithMultipleRules (#686)

The return value of `GetNetworkPolicy` is not deterministic as it
constructs rules from a map. Use `ElementsMatch` to ignore the order of
rules.

antoninbas committed:
Fix return path for NodePort Service traffic in EKS

When Antrea is used to enforce NetworkPolicies in EKS, an additional
iptables rule is required in the mangle table, to ensure that return
NodePort Service traffic which was load-balanced to a Pod attached to a
secondary ENI takes the correct reverse path through eth0.

This requires communicating to the Antrea Agent that it is being run in
an EKS managed cluster, which we do through an environment variable
(ANTREA_CLOUD_EKS).

Some future improvements are possible: this rule is only necessary if
NodePort is enabled in AWS VPC CNI, and in some rare cases it may be
possible for the packet mark to differ from the default value (0x80).

This patch also improves the test-conformance-eks.sh script which was
used for testing. In particular it ensures that the script can be used
on macOS.

Fixes #678

abhiraut committed:
Add lister and informer to codegen image (#688)

weiqiangt committed:
Update codegen image before generating (#696)
